### ISSUE TITLE
Modelling mode and Simulation mode

### ISSUE THREAD

**Issue by osemsem** [eGovlab/sense4us-simulation#21]
A clear distinction is needed between the two modes, (edit and simulate), use the names (modelling and simulation modes) instead, as follows:
- In the modelling mode the left panel shows the iconic buttons for creating model elements on the graphing canvas. - only a double click on a node or a link changes the left panel to node properties or link properties, a save/cancel button of the properties panel returns to the normal panel with the iconic buttons. 
- In the modelling mode the quantifications on the causal links (Transfer coefficients and time lags) appear on the links.
- In the simulation mode: the left panel shows the time step T (weeks/months/years) and the maximum number of iterations (n) to be entered by the user and a time sliders over time points, t= 0,T,2T, ..., nT
- In the simulation mode, an editable small table with a scroll bar appearing beside each origin showing the scenario of change and a non-editable percentage change appearing on each impact node that changes over time by moving the slider. As shown in the attached screenshot.
  ![labels](https://cloud.githubusercontent.com/assets/12909055/8699082/26692662-2b03-11e5-9300-691af756e983.png)

**Comment by Rhineheart** [comment]
The last point: you want it to show the change in real time as you move the slider, right? If so, is it okay to require the user to click on a button e.g. Simulate to calculate all the numbers for all the nodes in different time steps? Depending on the network, it might be delay between each time step as you drag the slider. If we just calculate everything (atleast the beginning) at once, we can setup the numbers to be fetched while sliding.

**Comment by osemsem** [comment]
Yes, it is Ok to calculate for all time points, store the simulation results and fetch them to be shown on the graph. The ones to be shown in a small table beside the nodes the changes defined by the user for origins.

**Comment by Rhineheart** [comment]
All of the controllable actors have positive changes, but only some are marked with a positive symbol (+). Why is this? I at first thought it was because it was a positive change from the last time step, but the actor at the bottom got no symbols at all. When should I show a symbol?

**Comment by osemsem** [comment]
Do you mean controllable independent variables (i.e., policy instruments)?
If so, then both positive and negative changes in their levels are considered, for a +ve change it is possible to use a (+) sign or not, preferably use a (+) sign, while for a -ve change, of course we must show a (-) sign.
The symbol for an actor associated to/ beside a variable means this actor is the owner or controller of the variable.

**Comment by Rhineheart** [comment]
Oh, yeah, the independent variables. So, if T0 is a positive change of 10%, it should be shown as: T0, + 10%. And if T1 is negative: T1, - 4%

**Comment by osemsem** [comment]
Yes, exactly.

**Comment by Rhineheart** [comment]
I've created an issue to align these properly, but http://memset.se/200/427ed7871567c46316d8

Disregard the icon used, and the text below. This is correct, yes?

**Comment by osemsem** [comment]
Yes, but what does the Zero under the icon mean?

**Comment by osemsem** [comment]
Aha. It must be the name of the variable under the icon, right?

**Comment by Rhineheart** [comment]
Disregard it. It's mockup.

**Comment by osemsem** [comment]
Ok

**Comment by Rhineheart** [comment]
Yeah it's the location for the name.

**Comment by Rhineheart** [comment]
Hmm, I didn't have scrolling and editing in mind when I created this table beside the independent variables. The easiest way to achieve this is to create DOM elements beside each node. But to not kill slow computers, I'd have to lock the panning of the canvas since moving and re-rendering a lot of DOM elements is pretty demanding. I mean, I could try but it would probably be slow as hell. Do you think we could maybe make the table selectable and you can change it on the side? I could create a table you edit in the canvas, but that feels like a huge waste of time when there are simpler solutions. Opinions? DOM elements? Sidebar?

http://memset.se/201/48bc0118ac701b6c6f08

It's now aligned a little bit better.
